model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro_program.py

```diff
@@ -337,6 +339,172 @@ def predictor_stencils_7_8_9(
     )
 
 
+@gtx.field_operator
+def _predictor_stencils_7_8_9_firststep(
+    rho: fa.CellKField[float],
+    z_rth_pr_1: fa.CellKField[float],
+    z_rth_pr_2: fa.CellKField[float],
+    rho_ref_mc: fa.CellKField[float],
+    theta_v: fa.CellKField[float],
+    theta_ref_mc: fa.CellKField[float],
+    rho_ic: fa.CellKField[float],
+    wgtfac_c: fa.CellKField[float],
+    vwind_expl_wgt: fa.CellField[float],
+    exner_pr: fa.CellKField[float],
+    d_exner_dz_ref_ic: fa.CellKField[float],
+    z_theta_v_pr_ic: fa.CellKField[float],
+    theta_v_ic: fa.CellKField[float],
+    k_field: fa.KField[gtx.int32],
+    nlev: gtx.int32,
+) -> tuple[
+    fa.CellKField[float],
+    fa.CellKField[float],
+    fa.CellKField[float],
+    fa.CellKField[float],
+    fa.CellKField[float],
+]:
+    (z_rth_pr_1, z_rth_pr_2) = where(
+        k_field == 0,
+        _compute_perturbation_of_rho_and_theta(rho, rho_ref_mc, theta_v, theta_ref_mc),
+        (z_rth_pr_1, z_rth_pr_2),
+    )
+
+    (rho_ic, z_rth_pr_1, z_rth_pr_2) = where(
+        k_field >= 1,
+        _compute_perturbation_of_rho_and_theta_and_rho_interface_cell_centers(
+            wgtfac_c, rho, rho_ref_mc, theta_v, theta_ref_mc
+        ),
+        (rho_ic, z_rth_pr_1, z_rth_pr_2),
+    )
+
+    (z_theta_v_pr_ic, theta_v_ic) = where(
+        k_field >= 1,
+        _compute_virtual_potential_temperatures(
+            wgtfac_c,
+            z_rth_pr_2,
+            theta_v,
+            d_exner_dz_ref_ic,
+        ),
+        (z_theta_v_pr_ic, theta_v_ic),
+    )
+
+    return z_rth_pr_1, z_rth_pr_2, rho_ic, z_theta_v_pr_ic, theta_v_ic
+
+
+@gtx.program(grid_type=gtx.GridType.UNSTRUCTURED, backend=backend)
+def predictor_stencils_7_8_9_firststep(
```

Comment:
I have another PR https://github.com/C2SM/icon4py/blob/e2dd88f2cdfa130baca336d473294feb28a592cb/model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/nh_solve/solve_nonhydro_program.py#L390 that also deals with this stencil in order to run the JW test on GPU. I renamed it so that it is more understandable what this stencil actually does. Maybe you can take a look and rename too.

Reply:
thanks, will do